llvm/unittests/Support/ErrorTest.cpp

```diff
@@ -1172,6 +1173,7 @@ TEST(Error, ForwardToExpected) {
   };
   std::optional<int> MaybeV;
   EXPECT_THAT_ERROR(ExpectedReturningFct(true).moveInto(MaybeV), Failed());
```

Comment:
Huh - how come this existing caller didn't catch the problem already? I thought the issue was that the code didn't compile with the older clang. In which case a fail/success case wouldn't change whether the "does this compile" was exercised, right?

Reply:
Error->Error forwarding worked. Error->Expected<T> didn't (if I understand your question right)

Reply:
Oh, sorry, think I was accidentally looking at a single commit (the last one) in the PR, rather than the whole PR diff. My mistake!